src/utils/BackendClient/NimbledroidApiHandler.js

```diff
@@ -110,6 +110,12 @@ class NimbledroidApiHandler {
     const productsData = await Promise.all(
       this.products.map(async product => this.fetchProductData(product)),
     );
+    if (
+      Object.keys(productsData[0]).length === 0 ||
+      Object.keys(productsData[1]).length === 0
+    ) {
+      throw Error('The backend has no data available.');
```

Comment:
I'm curious, why are you throwing an error message instead of returning an http status code to be handled by the frontend? If you need to do it this way, it'd be a good idea to have a comment explaining that any change in the message's wording needs to also be changed in the frontend or it'll break the logic.